packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -555,6 +556,13 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
                     if (clientIdOrDocumentId.length === 0 ||
                         clientIdOrDocumentId === this.documentId ||
                         (this.hasDetails && clientIdOrDocumentId === this.clientId)) {
+                        const nackContent = message[0]?.content;
+                        if (nackContent !== undefined) {
+                            this.logger.sendTelemetryEvent({
+                                eventName: "ServerNackOdsp",
+                                ...nackContent,
```

Comment:
The content is actually structured as interface:
export interface INackContent {
    /**
     * An error code number that represents the error. It will be a valid HTTP error code.
     * 403 errors are non retryable and client should acquire a new identity before reconnection.
     * 400 errors are always immediately retriable
     * 429 errors are retriable or non retriable (depends on type field).
     */
    code: number;

    /**
     * Type of the Nack.
     */
    type: NackErrorType;

    /**
     * A message about the nack for debugging/logging/telemetry purposes
     */
    message: string;

    /**
     * Optional Retry-After time in seconds
     * If specified, the client should wait this many seconds before retrying
     */
    retryAfter?: number;
}

Reply:
Do you think it would still be better to jsonify it and log in single field so that we don't miss any unknown fields that might be there?

Reply:
Made the change actually.

Reply:
Gotcha.  Even though we have it modeled with an interface on our side, who knows what will actually come over the wire, so better to play it safe.

Stringifying it as-is into the `details` field probably sounds good to me. The only other risk is privacy, if there were extra details in the object that contained user identifiers or data.  Sounds like we have reason to trust `message`, and hopefully there won't be anything else weird...

I would do something like this, or as discussed below, maybe each field does get its own column instead of using `details`:

```
const { code, type, message, retryAfter } = nackContent;
  ... 
  details: JSON.stringify({ code, type, message }),
  retryAfterMs: 1000 * retryAfter,
};
```

Things to check Kusto and think about for exactly what to do here:
* Is `retryAfterMs` or `retryAfter` a common field from other events? Pick whichever one seems to be more standard/common (and hopefully `retryAfter` doesn't have mixed units between seconds and ms... probably avoid it if so)
* Will `code`, `type` and `message` "stand alone" in telemetry if used as columns instead of combined into `details`?
  * `message` is sometimes (often) an error message.  This seems like it's probably ok to use `message` here
  * `code` maybe should be renamed to `statusCode` in the logs?  Again, take a look at what is most common.
  * `type` - Does this column exist, and what kind of info is in here from other events? Does the meaning of this field match?  If not maybe rename to `nackType` - maybe other nack-related logs are already doing something here.

The big idea is to make it more intuitive which columns to query and how to interpret them and having a better shot at a smaller subset of columns being relevant/reused in multiple events.  `details` is a new standard I'm starting to use to solve this in the "long tail" of events with super domain-specific props, but as discussed the info here might actually fit into existing semi-standard columns - just try to get the naming right to match any precedent.

Reply:
Logged separately with apt column names. We do code in telemetry and type too. We don't have nackType so didn't use that.